diff -ur kdebase-workspace-4.2.1.org/doc/kdm/index.docbook kdebase-workspace-4.2.1/doc/kdm/index.docbook
--- kdebase-workspace-4.2.1.org/doc/kdm/index.docbook	2008-05-29 13:18:56.000000000 +0200
+++ kdebase-workspace-4.2.1/doc/kdm/index.docbook	2009-03-01 13:33:49.650643144 +0100
@@ -317,7 +317,7 @@
 system, and that you simply want to change its behavior in some way.</para>
 
 <para>When &kdm; starts up, it reads its configuration from the folder
-<filename class="directory">$<envar>KDEDIR</envar>/share/config/kdm/</filename>
+<filename class="directory">/etc/X11/kdm/</filename>
 (this may
 be <filename class="directory">/etc/kde4/kdm/</filename> or something else
 on your system).</para>
diff -ur kdebase-workspace-4.2.1.org/kcontrol/fonts/fonts.cpp kdebase-workspace-4.2.1/kcontrol/fonts/fonts.cpp
--- kdebase-workspace-4.2.1.org/kcontrol/fonts/fonts.cpp	2008-12-21 21:19:15.000000000 +0100
+++ kdebase-workspace-4.2.1/kcontrol/fonts/fonts.cpp	2009-03-01 13:30:49.730658303 +0100
@@ -643,7 +643,7 @@
        "<p>The use of this option is generally discouraged. For selecting proper DPI"
        " value a better option is explicitly configuring it for the whole X server if"
        " possible (e.g. DisplaySize in xorg.conf or adding <i>-dpi value</i> to"
-       " ServerLocalArgs= in $KDEDIR/share/config/kdm/kdmrc). When fonts do not render"
+       " ServerLocalArgs= in /etc/X11/kdm/kdmrc). When fonts do not render"
        " properly with real DPI value better fonts should be used or configuration"
        " of font hinting should be checked.</p>" );
    comboForceDpi->setWhatsThis(whatsthis);
diff -ur kdebase-workspace-4.2.1.org/kcontrol/kdm/background.cpp kdebase-workspace-4.2.1/kcontrol/kdm/background.cpp
--- kdebase-workspace-4.2.1.org/kcontrol/kdm/background.cpp	2008-07-03 07:05:24.000000000 +0200
+++ kdebase-workspace-4.2.1/kcontrol/kdm/background.cpp	2009-03-01 13:34:38.617307679 +0100
@@ -45,7 +45,7 @@
 		" specified in the Setup= option in kdmrc (usually Xsetup).") );
 	m_simpleConf = KSharedConfig::openConfig(
 		config->group( "X-*-Greeter" ).readEntry(
-			"BackgroundCfg", KDE_CONFDIR "/kdm/backgroundrc" ) );
+			"BackgroundCfg", "/etc/X11/kdm/backgroundrc" ) );
 	m_background = new BGDialog( this, m_simpleConf );
 
 	connect( m_background, SIGNAL(changed( bool )), SIGNAL(changed()) );
diff -ur kdebase-workspace-4.2.1.org/kcontrol/kdm/kdm-users.cpp kdebase-workspace-4.2.1/kcontrol/kdm/kdm-users.cpp
--- kdebase-workspace-4.2.1.org/kcontrol/kdm/kdm-users.cpp	2008-08-28 18:12:26.000000000 +0200
+++ kdebase-workspace-4.2.1/kcontrol/kdm/kdm-users.cpp	2009-03-01 13:43:23.500581732 +0100
@@ -87,7 +87,7 @@
 
 	// We assume that $kde_datadir/kdm exists, but better check for pics/ and pics/users,
 	// and create them if necessary.
-	m_userPixDir = config->group( "X-*-Greeter" ).readEntry( "FaceDir", KStandardDirs::installPath( "data" ) + "kdm/faces" ) + '/';
+	m_userPixDir = config->group( "X-*-Greeter" ).readEntry( "FaceDir", "/etc/X11/kdm/faces" ) + '/';
 	QDir testDir( m_userPixDir );
 	if (!testDir.exists() && !testDir.mkdir( testDir.absolutePath() ) && !geteuid())
 		KMessageBox::sorry( this, i18n("Unable to create folder %1", testDir.absolutePath() ) );
diff -ur kdebase-workspace-4.2.1.org/kcontrol/kdm/main.cpp kdebase-workspace-4.2.1/kcontrol/kdm/main.cpp
--- kdebase-workspace-4.2.1.org/kcontrol/kdm/main.cpp	2009-01-06 18:22:15.000000000 +0100
+++ kdebase-workspace-4.2.1/kcontrol/kdm/main.cpp	2009-03-01 13:30:59.824472782 +0100
@@ -187,7 +187,7 @@
 		kWarning() << "user(s) '" << tgmapci.value().join( "," )
 		<< "' have unknown GID " << tgmapci.key() << endl;
 
-	config = new KConfig( QString::fromLatin1(KDE_CONFDIR "/kdm/kdmrc"), KConfig::SimpleConfig );
+	config = new KConfig( QString::fromLatin1("/etc/X11/kdm/kdmrc"), KConfig::SimpleConfig );
 
 	QVBoxLayout *top = new QVBoxLayout( this );
 	top->setMargin( 0 );
diff -ur kdebase-workspace-4.2.1.org/kdm/config.def kdebase-workspace-4.2.1/kdm/config.def
--- kdebase-workspace-4.2.1.org/kdm/config.def	2009-02-26 15:16:59.000000000 +0100
+++ kdebase-workspace-4.2.1/kdm/config.def	2009-03-01 13:42:27.887607315 +0100
@@ -29,7 +29,7 @@
 #define RCVERMAJOR 2
 #define RCVERMINOR 4
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF "/etc/X11/kdm"
 #define KDMDATA KDE_DATADIR "/kdm"
 
 #ifdef _AIX
@@ -1163,7 +1163,7 @@
 
 Key: Willing
 Type: string
-Default: ""
+Default: KDMCONF "/Xwilling"
 User: core
 # will be overwritten
 Instance: #
@@ -1602,7 +1602,7 @@
 
 Key: Startup
 Type: string
-Default: ""
+Default: KDMCONF "/Xsetup"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1620,7 +1620,7 @@
 
 Key: Reset
 Type: string
-Default: ""
+Default: KDMCONF "Xreset"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1638,7 +1638,7 @@
 
 Key: Session
 Type: string
-Default: XBINDIR "/xterm -ls -T"
+Default: KDMCONF "/Xsession"
 #Merge: xdm - incompatible!
 User: core
 # will be overwritten
@@ -1755,10 +1755,10 @@
 
 Key: AllowRootLogin
 Type: bool
-Default: true
+Default: false
 User: core
 User: greeter(showRoot)
-Instance: */false
+Instance: #*/true
 Merge: xdm
 Comment:
  Allow root logins?
@@ -1768,11 +1768,11 @@
 
 Key: AllowNullPasswd
 Type: bool
-Default: true
+Default: false
 User: core
 # sensible?
-Instance: */false
-Instance: :*/true
+Instance: #*/true
+Instance: #:*/true
 Merge: xdm
 Comment:
  Allow to log in, when user has set an empty password?
@@ -1930,10 +1930,10 @@
 
 Key: SessionsDirs
 Type: list
-Default: *KDMDATA "/sessions"
+Default: *"/usr/share/xsessions"
 User: core
 User: greeter-c
-Instance: */KDMCONF "/sessions," KDMDATA "/sessions"
+Instance: #*/"/etc/X11/sessions"
 Update: upd_sessionsdirs
 Comment:
  The directories containing session type definitions in .desktop format,
@@ -2277,7 +2277,7 @@
 
 Key: MinShowUID
 Type: int
-Default: 0
+Default: 1000
 User: greeter(lowUserId)
 # will be overwritten
 Instance: #*/
@@ -2338,7 +2338,7 @@
 
 Key: FaceDir
 Type: string
-Default: *KDMDATA "/faces"
+Default: *KDMCONF "/faces"
 User: greeter
 Instance: #*/"/usr/share/faces"
 Update: upd_facedir
diff -ur kdebase-workspace-4.2.1.org/kdm/kfrontend/CMakeLists.txt kdebase-workspace-4.2.1/kdm/kfrontend/CMakeLists.txt
--- kdebase-workspace-4.2.1.org/kdm/kfrontend/CMakeLists.txt	2008-07-31 10:14:09.000000000 +0200
+++ kdebase-workspace-4.2.1/kdm/kfrontend/CMakeLists.txt	2009-03-01 13:37:39.710759801 +0100
@@ -109,5 +109,5 @@
 
 ## use 'GENKDMCONF_FLAGS=... make install' to add flags to the config generation (try genkdmconf -h)
 install(CODE "
-exec_program(\"${CMAKE_CURRENT_BINARY_DIR}/genkdmconf\" ARGS --in \\\"\\\$DESTDIR${CONFIG_INSTALL_DIR}/kdm\\\" --no-in-notice --face-src \\\"${CMAKE_CURRENT_SOURCE_DIR}/pics\\\" \\\$GENKDMCONF_FLAGS)
+exec_program(\"${CMAKE_CURRENT_BINARY_DIR}/genkdmconf\" ARGS --in \\\"\\\${DESTDIR}/etc/X11/kdm\\\" --no-in-notice --face-src \\\"${CMAKE_CURRENT_SOURCE_DIR}/pics\\\" \\\$GENKDMCONF_FLAGS)
 ")
diff -ur kdebase-workspace-4.2.1.org/kdm/kfrontend/kdmctl.c kdebase-workspace-4.2.1/kdm/kfrontend/kdmctl.c
--- kdebase-workspace-4.2.1.org/kdm/kfrontend/kdmctl.c	2008-01-05 00:55:45.000000000 +0100
+++ kdebase-workspace-4.2.1/kdm/kfrontend/kdmctl.c	2009-03-01 13:31:07.834004939 +0100
@@ -150,7 +150,7 @@
 {
 	char *dpy = getenv( "DISPLAY" );
 	const char *ctl = getenv( "DM_CONTROL" );
-	const char *cfg = KDE_CONFDIR "/kdm/kdmrc";
+	const char *cfg = "/etc/X11/kdm/kdmrc";
 	char *ptr;
 	int fd;
 
--- kdebase-workspace-4.2.1/kdm/kfrontend/sessions/CMakeLists.txt.orig	2009-03-02 01:19:34.258648722 +0100
+++ kdebase-workspace-4.2.1/kdm/kfrontend/sessions/CMakeLists.txt	2009-03-02 01:20:25.502777197 +0100
@@ -51,5 +51,5 @@
 	xfce.desktop
 	xfce4.desktop
 
-	DESTINATION ${DATA_INSTALL_DIR}/kdm/sessions
+	DESTINATION /etc/X11/sessions
 )
