--- kdebase-workspace-4.3.4/CMakeLists.txt~	2009-11-27 14:01:58.000000000 +0100
+++ kdebase-workspace-4.3.4/CMakeLists.txt	2010-01-05 14:23:22.615963105 +0100
@@ -59,10 +59,7 @@
 endif(PYQT4_FOUND AND PYQT4_VERSION STRLESS "040400")
 
 macro_optional_find_package(PyKDE4)
-set(PythonSupport_FOUND FALSE)
-if(PYTHONLIBRARY_FOUND AND SIP_FOUND AND PYQT4_FOUND AND PYKDE4_FOUND)
-  set(PythonSupport_FOUND TRUE)
-endif(PYTHONLIBRARY_FOUND AND SIP_FOUND AND PYQT4_FOUND AND PYKDE4_FOUND)
+set(PythonSupport_FOUND TRUE)
 
 macro_log_feature(PythonSupport_FOUND "Python" "KDE Python support" "http://techbase.kde.org/Development/Languages/Python" FALSE "" "Needed to build plasma scriptengine for Python")
 #### End Python support ####
