diff -Nru kdebase-workspace-4.3.73svn1040395/doc/kdm/index.docbook kdebase-workspace-4.3.73svn1040395-new/doc/kdm/index.docbook
--- kdebase-workspace-4.3.73svn1040395/doc/kdm/index.docbook	2009-10-26 10:14:41.000000000 +0100
+++ kdebase-workspace-4.3.73svn1040395-new/doc/kdm/index.docbook	2009-10-28 22:58:07.000000000 +0100
@@ -317,7 +317,7 @@
 system, and that you simply want to change its behavior in some way.</para>
 
 <para>When &kdm; starts up, it reads its configuration from the folder
-<filename class="directory">$<envar>KDEDIR</envar>/share/config/kdm/</filename>
+<filename class="directory">/etc/X11/kdm/</filename>
 (this may
 be <filename class="directory">/etc/kde4/kdm/</filename> or something else
 on your system).</para>
diff -Nru kdebase-workspace-4.3.73svn1040395/kcontrol/fonts/fonts.cpp kdebase-workspace-4.3.73svn1040395-new/kcontrol/fonts/fonts.cpp
--- kdebase-workspace-4.3.73svn1040395/kcontrol/fonts/fonts.cpp	2009-10-02 16:53:57.000000000 +0200
+++ kdebase-workspace-4.3.73svn1040395-new/kcontrol/fonts/fonts.cpp	2009-10-28 22:58:08.000000000 +0100
@@ -638,7 +638,7 @@
        "<p>The use of this option is generally discouraged. For selecting proper DPI"
        " value a better option is explicitly configuring it for the whole X server if"
        " possible (e.g. DisplaySize in xorg.conf or adding <i>-dpi value</i> to"
-       " ServerLocalArgs= in $KDEDIR/share/config/kdm/kdmrc). When fonts do not render"
+       " ServerLocalArgs= in /etc/X11/kdm/kdmrc). When fonts do not render"
        " properly with real DPI value better fonts should be used or configuration"
        " of font hinting should be checked.</p>" );
    comboForceDpi->setWhatsThis(whatsthis);
--- kdebase-workspace-4.5.0/kcontrol/kdm/kdm-users.cpp.orig	2010-05-27 19:23:09.000000000 +0200
+++ kdebase-workspace-4.5.0/kcontrol/kdm/kdm-users.cpp	2010-07-30 20:37:06.492035540 +0200
@@ -103,8 +103,7 @@
     defmaxuid = "65000";
 #endif
 
-    m_userPixDir = config->group("X-*-Greeter").readEntry("FaceDir",
-            KStandardDirs::installPath("data") + "kdm/faces") + '/';
+    m_userPixDir = config->group( "X-*-Greeter" ).readEntry( "FaceDir", "/etc/X11/kdm/faces" ) + '/';
 
     if (!getpwnam("nobody"))
         KMessageBox::sorry(this, i18n(
diff -Nru kdebase-workspace-4.3.73svn1040395/kdm/config.def kdebase-workspace-4.3.73svn1040395-new/kdm/config.def
--- kdebase-workspace-4.3.73svn1040395/kdm/config.def	2009-10-26 10:14:48.000000000 +0100
+++ kdebase-workspace-4.3.73svn1040395-new/kdm/config.def	2009-10-28 22:58:08.000000000 +0100
@@ -29,7 +29,7 @@
 #define RCVERMAJOR 2
 #define RCVERMINOR 4
 
-#define KDMCONF KDE_CONFDIR "/kdm"
+#define KDMCONF "/etc/X11/kdm"
 #define KDMDATA KDE_DATADIR "/kdm"
 
 #ifdef _AIX
@@ -1099,7 +1099,7 @@
 Key: Xaccess
 Type: string
 # differs from xdm
-Default: *KDMCONF "/Xaccess"
+Default: *"/etc/X11/kdm/Xaccess"
 User: config(Xaccess)
 Instance: #""
 Update: mk_xaccess
@@ -1161,7 +1161,7 @@
 
 Key: Willing
 Type: string
-Default: ""
+Default: KDMCONF "/Xwilling"
 User: core
 # will be overwritten
 Instance: #
@@ -1600,7 +1600,7 @@
 
 Key: Startup
 Type: string
-Default: ""
+Default: KDMCONF "/Xsetup"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1618,7 +1618,7 @@
 
 Key: Reset
 Type: string
-Default: ""
+Default: KDMCONF "Xreset"
 User: core
 # will be overwritten
 Instance: #*/""
@@ -1636,7 +1636,7 @@
 
 Key: Session
 Type: string
-Default: XBINDIR "/xterm -ls -T"
+Default: KDMCONF "/Xsession"
 #Merge: xdm - incompatible!
 User: core
 # will be overwritten
@@ -1753,10 +1753,10 @@
 
 Key: AllowRootLogin
 Type: bool
-Default: true
+Default: false
 User: core
 User: greeter(showRoot)
-Instance: */false
+Instance: #*/true
 Merge: xdm
 Comment:
  Allow root logins?
@@ -1766,11 +1766,11 @@
 
 Key: AllowNullPasswd
 Type: bool
-Default: true
+Default: false
 User: core
 # sensible?
-Instance: */false
-Instance: :*/true
+Instance: #*/true
+Instance: #:*/true
 Merge: xdm
 Comment:
  Allow to log in, when user has set an empty password?
@@ -1928,10 +1928,10 @@
 
 Key: SessionsDirs
 Type: list
-Default: *KDMDATA "/sessions"
+Default: *"/usr/share/xsessions"
 User: core
 User: greeter-c
-Instance: */KDMCONF "/sessions," KDMDATA "/sessions"
+Instance: */"/etc/X11/sessions,/usr/share/xsessions"
 Update: upd_sessionsdirs
 Comment:
  The directories containing session type definitions in .desktop format,
@@ -2275,7 +2275,7 @@
 
 Key: MinShowUID
 Type: int
-Default: 0
+Default: 1000
 User: greeter(lowUserId)
 # will be overwritten
 Instance: #*/
@@ -2336,7 +2336,7 @@
 
 Key: FaceDir
 Type: string
-Default: *KDMDATA "/faces"
+Default: *"/etc/X11/kdm/faces"
 User: greeter
 Instance: #*/"/usr/share/faces"
 Update: upd_facedir
@@ -2419,7 +2419,7 @@
 
 Key: BackgroundCfg
 Type: string
-Default: *KDMCONF "/backgroundrc"
+Default: *"/etc/X11/kdm/backgroundrc"
 User: greeter-c
 Instance: #*/""
 Update: handleBgCfg
diff -Nru kdebase-workspace-4.3.73svn1040395/kdm/kfrontend/CMakeLists.txt kdebase-workspace-4.3.73svn1040395-new/kdm/kfrontend/CMakeLists.txt
--- kdebase-workspace-4.3.73svn1040395/kdm/kfrontend/CMakeLists.txt	2009-10-02 16:54:04.000000000 +0200
+++ kdebase-workspace-4.3.73svn1040395-new/kdm/kfrontend/CMakeLists.txt	2009-10-28 22:58:08.000000000 +0100
@@ -112,5 +112,5 @@
 
 ## use 'GENKDMCONF_FLAGS=... make install' to add flags to the config generation (try genkdmconf -h)
 install(CODE "
-exec_program(\"${CMAKE_CURRENT_BINARY_DIR}/genkdmconf\" ARGS --in \\\"\\\$DESTDIR${CONFIG_INSTALL_DIR}/kdm\\\" --no-in-notice --face-src \\\"${CMAKE_CURRENT_SOURCE_DIR}/pics\\\" \\\$GENKDMCONF_FLAGS)
+exec_program(\"${CMAKE_CURRENT_BINARY_DIR}/genkdmconf\" ARGS --in \\\"\\\${DESTDIR}/etc/X11/kdm\\\" --no-in-notice --face-src \\\"${CMAKE_CURRENT_SOURCE_DIR}/pics\\\" \\\$GENKDMCONF_FLAGS)
 ")
--- kdebase-workspace-4.5.0/kdm/kfrontend/genkdmconf.c.orig	2010-07-06 08:54:31.000000000 +0200
+++ kdebase-workspace-4.5.0/kdm/kfrontend/genkdmconf.c	2010-07-30 20:40:48.828680855 +0200
@@ -3254,7 +3254,7 @@
     f = createFile("README", 0644);
     fprintf_(f,
 "This automatically generated configuration consists of the following files:\n");
-    fprintf_(f, "- " KDMCONF "/kdmrc\n");
+    fprintf( f, "- /etc/X11/kdm/kdmrc\n" );
     for (fp = aflist; fp; fp = fp->next)
         fprintf_(f, "- %s\n", fp->str);
     if (use_destdir && !no_in_notice)
--- kdebase-workspace-4.5.0/kdm/kfrontend/kdmctl.c.orig	2010-07-06 08:54:31.000000000 +0200
+++ kdebase-workspace-4.5.0/kdm/kfrontend/kdmctl.c	2010-07-30 20:42:26.855824153 +0200
@@ -150,7 +150,7 @@
 {
     char *dpy = getenv("DISPLAY");
     const char *ctl = getenv("DM_CONTROL");
-    const char *cfg = KDE_CONFDIR "/kdm/kdmrc";
+    const char *cfg = "/etc/X11/kdm/kdmrc";
     char *ptr;
     int fd;
 
diff -Nru kdebase-workspace-4.3.73svn1040395/kdm/kfrontend/kdmctl.c kdebase-workspace-4.3.73svn1040395-new/kdm/kfrontend/kdmctl.c
--- kdebase-workspace-4.3.73svn1040395/kdm/kfrontend/kdmctl.c	2008-01-05 00:55:45.000000000 +0100
+++ kdebase-workspace-4.3.73svn1040395-new/kdm/kfrontend/kdmctl.c	2009-10-28 22:58:08.000000000 +0100
@@ -150,7 +150,7 @@
 {
 	char *dpy = getenv( "DISPLAY" );
 	const char *ctl = getenv( "DM_CONTROL" );
-	const char *cfg = KDE_CONFDIR "/kdm/kdmrc";
+	const char *cfg = "/etc/X11/kdm/kdmrc";
 	char *ptr;
 	int fd;
 
diff -Nru kdebase-workspace-4.3.73svn1040395/kdm/kfrontend/sessions/CMakeLists.txt kdebase-workspace-4.3.73svn1040395-new/kdm/kfrontend/sessions/CMakeLists.txt
--- kdebase-workspace-4.3.73svn1040395/kdm/kfrontend/sessions/CMakeLists.txt	2008-04-30 15:58:40.000000000 +0200
+++ kdebase-workspace-4.3.73svn1040395-new/kdm/kfrontend/sessions/CMakeLists.txt	2009-10-28 22:58:08.000000000 +0100
@@ -51,5 +51,5 @@
 	xfce.desktop
 	xfce4.desktop
 
-	DESTINATION ${DATA_INSTALL_DIR}/kdm/sessions
+	DESTINATION /etc/X11/sessions
 )
