diff -urN kde-workspace-4.9.4.org/powerdevil/daemon/backends/upower/powerdevilupowerbackend.cpp kde-workspace-4.9.4/powerdevil/daemon/backends/upower/powerdevilupowerbackend.cpp
--- kde-workspace-4.9.4.org/powerdevil/daemon/backends/upower/powerdevilupowerbackend.cpp	2012-11-30 08:24:40.000000000 +0100
+++ kde-workspace-4.9.4/powerdevil/daemon/backends/upower/powerdevilupowerbackend.cpp	2012-12-02 11:58:03.781622657 +0100
@@ -165,9 +165,9 @@
     if (qFuzzyCompare(currentBrightness, m_cachedBrightness)) {
         float newBrightness;
         if (type == Increase) {
-            newBrightness = qMin(100.0f, currentBrightness + 10);
+            newBrightness = qMin(100.0f, currentBrightness + 6);
         } else {
-            newBrightness = qMax(0.0f, currentBrightness - 10);
+            newBrightness = qMax(0.0f, currentBrightness - 6);
         }
 
         setBrightness(newBrightness, Screen);
