--- workspace/kcontrol/input/logitechmouse.cpp	(.../tags/KDE/4.3.4/kdebase)	(wersja 1057516)
+++ workspace/kcontrol/input/logitechmouse.cpp	(.../branches/KDE/4.3/kdebase)	(wersja 1057516)
@@ -38,6 +38,7 @@
 
 LogitechMouse::LogitechMouse( struct usb_device *usbDev, int mouseCapabilityFlags, QWidget* parent, const char* name )
     : LogitechMouseBase( parent )
+    , m_resolution( 0 )
 {
     if ( !name )
         setObjectName( "LogitechMouse" );
@@ -110,7 +111,8 @@
 
 LogitechMouse::~LogitechMouse()
 {
-    usb_close( m_usbDeviceHandle );
+    if ( m_usbDeviceHandle )
+        usb_close( m_usbDeviceHandle );
 }
 
 void LogitechMouse::initCordlessStatusReporting()
@@ -125,7 +127,10 @@
 {
     QByteArray status(8, '\0');
 
-    int result =  usb_control_msg(  m_usbDeviceHandle,
+    int result = -1;
+
+    if ( m_usbDeviceHandle )
+        result = usb_control_msg( m_usbDeviceHandle,
                                     USB_TYPE_VENDOR | USB_ENDPOINT_IN,0x09,
                                     (0x0003 | m_useSecondChannel),
                                     (0x0000 | m_useSecondChannel),
@@ -135,6 +140,7 @@
 
     if (0 > result) {
         // We probably have a permission problem
+        m_channel = 0;
         channelSelector->setEnabled( false );
         batteryBox->setEnabled( false );
         cordlessNameLabel->hide();
@@ -246,7 +252,10 @@
 {
     char resolution;
 
-    int result =  usb_control_msg( m_usbDeviceHandle,
+    int result = -1;
+
+    if ( m_usbDeviceHandle )
+        result = usb_control_msg( m_usbDeviceHandle,
                                    USB_TYPE_VENDOR | USB_ENDPOINT_IN,
                                    0x01,
                                    0x000E,
--- workspace/kwin/effects/presentwindows/presentwindows.desktop	(.../tags/KDE/4.3.4/kdebase)	(wersja 1057516)
+++ workspace/kwin/effects/presentwindows/presentwindows.desktop	(.../branches/KDE/4.3/kdebase)	(wersja 1057516)
@@ -42,7 +42,7 @@
 Name[ml]=നിലവിലുള്ള ജാലകങ്ങള്‍
 Name[mr]=वर्तमान चौकट
 Name[nb]=Vis vinduer
-Name[nds]=Finstern wiesen
+Name[nds]=Vörhannen Finstern
 Name[ne]=हालको सञ्झ्याल
 Name[nl]=Vensters presenteren
 Name[nn]=Presenter vindauge
--- workspace/kwin/effects/presentwindows/presentwindows_config.desktop	(.../tags/KDE/4.3.4/kdebase)	(wersja 1057516)
+++ workspace/kwin/effects/presentwindows/presentwindows_config.desktop	(.../branches/KDE/4.3/kdebase)	(wersja 1057516)
@@ -49,7 +49,7 @@
 Name[ml]=നിലവിലുള്ള ജാലകങ്ങള്‍
 Name[mr]=वर्तमान चौकट
 Name[nb]=Vis vinduer
-Name[nds]=Finstern wiesen
+Name[nds]=Vörhannen Finstern
 Name[ne]=हालको सञ्झ्याल
 Name[nl]=Vensters presenteren
 Name[nn]=Presenter vindauge
--- workspace/plasma/applets/kickoff/simpleapplet/plasma-applet-simplelauncher.desktop	(.../tags/KDE/4.3.4/kdebase)	(wersja 1057516)
+++ workspace/plasma/applets/kickoff/simpleapplet/plasma-applet-simplelauncher.desktop	(.../branches/KDE/4.3/kdebase)	(wersja 1057516)
@@ -109,7 +109,7 @@
 Comment[ml]=പരമ്പരാഗതമായ മെനുഅടിസ്ഥാനമാക്കിയുള്ള പ്രയോഗവിക്ഷേപിണി
 Comment[mr]=पारंपारिक मेन्यू आधारीत अनुप्रयोग प्रक्षेपक
 Comment[nb]=Tradisjonell menybasert programstarter
-Comment[nds]=Begäng Programmstarter mit Menüs
+Comment[nds]=Begäng Programmoproper mit Menüs
 Comment[nl]=Traditioneel menu voor het starten van programma's
 Comment[nn]=Tradisjonell menybasert programstartar
 Comment[or]=ପ୍ରଥାଗତ ତାଲିକା ଆଧାରିତ ପ୍ରୟୋଗ ଆରମ୍ଭକର୍ତ୍ତା
--- workspace/plasma/applets/kickoff/applet/plasma-applet-launcher.desktop	(.../tags/KDE/4.3.4/kdebase)	(wersja 1057516)
+++ workspace/plasma/applets/kickoff/applet/plasma-applet-launcher.desktop	(.../branches/KDE/4.3/kdebase)	(wersja 1057516)
@@ -46,7 +46,7 @@
 Name[ml]=പ്രയോഗവിക്ഷേപിണി
 Name[mr]=अनुप्रयोग प्रकक्षेपक
 Name[nb]=Programstarter
-Name[nds]=Programmstarter
+Name[nds]=Programmoproper
 Name[ne]=अनुप्रयोक सुरुआतकर्ता
 Name[nl]=Programmastarter
 Name[nn]=Programstartar
