diff -Nur kdebase-workspace-4.0.74.orig/khotkeys/libkhotkeysprivate/CMakeLists.txt kdebase-workspace-4.0.74.chng/khotkeys/libkhotkeysprivate/CMakeLists.txt
--- kdebase-workspace-4.0.74.orig/khotkeys/libkhotkeysprivate/CMakeLists.txt	2008-04-10 11:30:15.000000000 +0200
+++ kdebase-workspace-4.0.74.chng/khotkeys/libkhotkeysprivate/CMakeLists.txt	2008-05-25 01:50:58.000000000 +0200
@@ -83,7 +83,7 @@
 )
 
 kde4_add_library(khotkeysprivate SHARED ${khotkeysprivate_SRCS})
-target_link_libraries(khotkeysprivate kworkspace ${KDE4_KDE3SUPPORT_LIBS})
+target_link_libraries(khotkeysprivate kworkspace ${KDE4_KDE3SUPPORT_LIBS} Xtst)
 set_target_properties(khotkeysprivate PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION})
 install(TARGETS khotkeysprivate DESTINATION ${LIB_INSTALL_DIR})
 
