--- kdebase-workspace-4.1.80/plasma/scriptengines/google_gadgets/ggl_package.cpp.orig	2008-11-19 11:18:19.000000000 +0100
+++ kdebase-workspace-4.1.80/plasma/scriptengines/google_gadgets/ggl_package.cpp	2008-11-27 15:44:44.000000000 +0100
@@ -63,11 +63,11 @@
         ggadget::BuildFilePath(ggadget::GetHomeDirectory().c_str(),
                                ".google/gadgets-plasma", NULL);
 
-    std::string error;
+    QString error;
     if (!ggadget::qt::InitGGL(NULL, "ggl-plasma", profile_dir.c_str(),
                               kGlobalExtensions, 0, false, &error)) {
       kError() << "Failed to init GGL system:"
-               << QString::fromUtf8(error.c_str());
+               << error;
       return;
     }
     gadget_manager_ = GetGadgetManager();
--- kdebase-workspace-4.1.80/plasma/scriptengines/google_gadgets/ggl_applet_script.cpp.orig	2008-11-19 11:18:19.000000000 +0100
+++ kdebase-workspace-4.1.80/plasma/scriptengines/google_gadgets/ggl_applet_script.cpp	2008-11-27 15:50:17.000000000 +0100
@@ -89,11 +89,11 @@
           ggadget::BuildFilePath(ggadget::GetHomeDirectory().c_str(),
                                  ".google/gadgets-plasma", NULL);
 
-  std::string error;
+  QString error;
   if (!ggadget::qt::InitGGL(NULL, "ggl-plasma", profile_dir.c_str(),
                             kGlobalExtensions, 0, false, &error)) {
     kError() << "Failed to init GGL system:"
-             << QString::fromUtf8(error.c_str());
+             << error;
     return false;
   }
 
--- kdebase-workspace-4.1.80/plasma/scriptengines/google_gadgets/plasma_view_host.h.orig	2008-11-19 11:18:19.000000000 +0100
+++ kdebase-workspace-4.1.80/plasma/scriptengines/google_gadgets/plasma_view_host.h	2008-11-27 16:22:18.000000000 +0100
@@ -67,7 +67,7 @@
   virtual void BeginMoveDrag(int) {}
 
   virtual void Alert(const ViewInterface *view, const char *message);
-  virtual bool Confirm(const ViewInterface *view, const char *message);
+  virtual ConfirmResponse Confirm(const ViewInterface *view, const char *message, bool cancel_button);
   virtual std::string Prompt(const ViewInterface *view,
                              const char *message,
                              const char *default_value);
--- kdebase-workspace-4.1.80/plasma/scriptengines/google_gadgets/plasma_view_host.cpp.orig	2008-11-19 11:18:19.000000000 +0100
+++ kdebase-workspace-4.1.80/plasma/scriptengines/google_gadgets/plasma_view_host.cpp	2008-11-27 16:27:45.000000000 +0100
@@ -168,11 +168,12 @@
                            view->GetCaption().c_str());
 }
 
-bool PlasmaViewHost::Confirm(const ViewInterface *view, const char *message) {
+ViewHostInterface::ConfirmResponse PlasmaViewHost::Confirm(const ViewInterface *view, const char *message, bool cancel_button) {
   int ret = KMessageBox::questionYesNo(NULL,
                                        message,
                                        view->GetCaption().c_str() );
-  return ret == KMessageBox::Yes;
+  return ret == KMessageBox::Yes ? CONFIRM_YES : 
+	 ret == KMessageBox::No ? CONFIRM_NO : cancel_button ? CONFIRM_CANCEL : CONFIRM_NO;
 }
 
 std::string PlasmaViewHost::Prompt(const ViewInterface *view,
