diff -ur kdebase-workspace-4.0.74/solid/networkmanager-0.6/NetworkManager-networkmanager.h kdebase-workspace-4.0.74.new/solid/networkmanager-0.6/NetworkManager-networkmanager.h
--- kdebase-workspace-4.0.74/solid/networkmanager-0.6/NetworkManager-networkmanager.h	2008-05-15 18:37:15.000000000 +0000
+++ kdebase-workspace-4.0.74.new/solid/networkmanager-0.6/NetworkManager-networkmanager.h	2008-05-25 14:09:47.000000000 +0000
@@ -29,7 +29,7 @@
 class NMNetworkManagerPrivate;
 class KDE_EXPORT NMNetworkManager : public Solid::Control::Ifaces::NetworkManager
 {
-Q_INTERFACES(Solid::Control::Ifaces::NetworkManager)
+//Q_INTERFACES(Solid::Control::Ifaces::NetworkManager)
 Q_OBJECT
     public:
         NMNetworkManager(QObject * parent, const QVariantList  & args);
Only in kdebase-workspace-4.0.74.new/solid/networkmanager-0.6: NetworkManager-networkmanager.h~
diff -ur kdebase-workspace-4.0.74/solid/networkmanager-0.7/manager.h kdebase-workspace-4.0.74.new/solid/networkmanager-0.7/manager.h
--- kdebase-workspace-4.0.74/solid/networkmanager-0.7/manager.h	2008-05-15 18:37:15.000000000 +0000
+++ kdebase-workspace-4.0.74.new/solid/networkmanager-0.7/manager.h	2008-05-25 14:34:43.000000000 +0000
@@ -32,7 +32,7 @@
 {
 Q_OBJECT
 Q_DECLARE_PRIVATE(NMNetworkManager)
-Q_INTERFACES(Solid::Control::Ifaces::NetworkManager)
+//Q_INTERFACES(Solid::Control::Ifaces::NetworkManager)
 
 public:
     static const QString DBUS_SERVICE;
Only in kdebase-workspace-4.0.74.new/solid/networkmanager-0.7: manager.h~
