--- workspace/ConfigureChecks.cmake	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/ConfigureChecks.cmake	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -85,3 +85,4 @@
 
 pkg_check_modules (XRANDR_1_2   xrandr>=1.2)
 macro_bool_to_01(XRANDR_1_2_FOUND HAS_RANDR_1_2)
+
--- workspace/klipper/klipperrc.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/klipper/klipperrc.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -259,7 +259,7 @@
 Description[gu]=મૂળભૂત બ્રાઉઝર સાથે ખોલો (&d)
 Description[he]=פתח באמצעות דפדפן &ברירת המחדל
 Description[hi]=डिफ़ॉल्ट ब्राउज़र खोलें (&d)
-Description[hne]=डिफाल्ट ब्राउजर के साथ खोलव (&d)
+Description[hne]=डिफाल्ट ब्राउजर एखर साथ खोलव (&d)
 Description[hr]=Otvori pomoću &zadanog preglednika
 Description[hsb]=Ze &standardnym browserom wočinić
 Description[hu]=Megnyitás az &alapértelmezett böngészővel
@@ -337,7 +337,7 @@
 Description[gu]=કોન્કરરની સાથે ખોલો (&K)
 Description[he]=פתח באמצעות &Konqueror
 Description[hi]=कॉन्करर के साथ खोलें (&K)
-Description[hne]=कान्करर के साथ खोलव (&K)
+Description[hne]=कान्करर एखर साथ खोलव (&K)
 Description[hr]=Otvori pomoću &Konquerora
 Description[hsb]=Z &konquerororm wočinić
 Description[hu]=Megnyitás a &Konquerorral
@@ -423,7 +423,7 @@
 Description[gu]=મોઝિલાની સાથે ખોલો (&M)
 Description[he]=פתח באמצעות &Mozilla
 Description[hi]=मोज़िला के साथ खोलें (&M)
-Description[hne]=मोजिला के साथ खोलव (&M)
+Description[hne]=मोजिला एखर साथ खोलव (&M)
 Description[hr]=Otvori pomoću &Mozille
 Description[hsb]=Z &Mozilla wočinić
 Description[hu]=Megnyitás a &Mozillával
@@ -592,7 +592,7 @@
 Description[gu]=ફાયરફોક્સની સાથે ખોલો (&F)
 Description[he]=פתח באמצעות &Firefox
 Description[hi]=फ़ॉयरफ़ॉक्स के साथ खोलें (&F)
-Description[hne]=फायरफाक्स के साथ खोलव (&F)
+Description[hne]=फायरफाक्स एखर साथ खोलव (&F)
 Description[hr]=Otvori pomoću &Firefoxa
 Description[hsb]=Z &Firefoxom wočinić
 Description[hu]=Megnyitás a &Firefoxszal
@@ -1724,7 +1724,7 @@
 Description[gu]=મૂળભૂત બ્રાઉઝર સાથે ખોલો (&d)
 Description[he]=פתח באמצעות דפדפן &ברירת המחדל
 Description[hi]=डिफ़ॉल्ट ब्राउज़र खोलें (&d)
-Description[hne]=डिफाल्ट ब्राउजर के साथ खोलव (&d)
+Description[hne]=डिफाल्ट ब्राउजर एखर साथ खोलव (&d)
 Description[hr]=Otvori pomoću &zadanog preglednika
 Description[hsb]=Ze &standardnym browserom wočinić
 Description[hu]=Megnyitás az &alapértelmezett böngészővel
@@ -1802,7 +1802,7 @@
 Description[gu]=કોન્કરરની સાથે ખોલો (&K)
 Description[he]=פתח באמצעות &Konqueror
 Description[hi]=कॉन्करर के साथ खोलें (&K)
-Description[hne]=कान्करर के साथ खोलव (&K)
+Description[hne]=कान्करर एखर साथ खोलव (&K)
 Description[hr]=Otvori pomoću &Konquerora
 Description[hsb]=Z &konquerororm wočinić
 Description[hu]=Megnyitás a &Konquerorral
@@ -1887,7 +1887,7 @@
 Description[gu]=મોઝિલાની સાથે ખોલો (&M)
 Description[he]=פתח באמצעות &Mozilla
 Description[hi]=मोज़िला के साथ खोलें (&M)
-Description[hne]=मोजिला के साथ खोलव (&M)
+Description[hne]=मोजिला एखर साथ खोलव (&M)
 Description[hr]=Otvori pomoću &Mozille
 Description[hsb]=Z &Mozilla wočinić
 Description[hu]=Megnyitás a &Mozillával
--- workspace/powerdevil/powerdevil.notifyrc	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/powerdevil/powerdevil.notifyrc	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -1270,6 +1270,7 @@
 Name[pl]=Przechodzenie do stanu uśpienia
 Name[pt]=A efectuar uma tarefa de suspensão
 Name[pt_BR]=Realizando uma tarefa de suspensão
+Name[ro]=Îndeplinire sarcină de suspendare
 Name[ru]=Переход в режим ожидания
 Name[sk]=Vykonávam uspanie
 Name[sl]=Izvajanje zaustavitvenega opravila
--- workspace/libs/solid/CMakeLists.txt	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/libs/solid/CMakeLists.txt	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -1,2 +1,4 @@
-add_subdirectory( control )
+if(NOT WIN32)
+  add_subdirectory( control )
+endif(NOT WIN32)
 
--- workspace/khotkeys/kcm_hotkeys/hotkeys_tree_view.cpp	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/khotkeys/kcm_hotkeys/hotkeys_tree_view.cpp	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -200,7 +200,8 @@
         Q_ASSERT( _index.isValid() );
         return;
         }
-    Q_ASSERT( _view->model()->removeRow(_index.row(), _index.parent()) );
+    bool success = _view->model()->removeRow(_index.row(), _index.parent());
+    Q_ASSERT(success);
 
     _view->setCurrentIndex(QModelIndex());
     }
--- workspace/khotkeys/data/konqueror_gestures_kde321.khotkeys	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/khotkeys/data/konqueror_gestures_kde321.khotkeys	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -1026,6 +1026,7 @@
 Comment[pl]=Naciśnij, w dół, w prawo, puść.
 Comment[pt]=Carregar, mover para cima, mover para a direita, largar.
 Comment[pt_BR]=Pressionar, mover para cima, mover para a direita, largar.
+Comment[ro]=Apasă, mută în sus, mută în deapta, eliberează.
 Comment[ru]=Нажать, вверх, вправо, отпустить.
 Comment[se]=Deaddil, sirdde bajásguvlui, sirdde olgešguvlui, luoitte.
 Comment[sk]=Stlačte, pohyb nahor, pohyb vpravo, pustite.
--- workspace/khotkeys/data/kde32b1.khotkeys	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/khotkeys/data/kde32b1.khotkeys	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -756,7 +756,7 @@
 Name[pl]=Uruchom Konsolę
 Name[pt]=Executar o Konsole
 Name[pt_BR]=Executar o Konsole
-Name[ro]=Lansează consolă
+Name[ro]=Lansează konsola
 Name[ru]=Запустить Konsole
 Name[se]=Vuoje konsolla
 Name[sk]=Spustiť Konsole
@@ -930,7 +930,7 @@
 Name[ml]=Qt ഡിസൈനറില്‍ Ctrl+W to Ctrl+F4 വീണ്ടും മാപ്പ് ചെയ്യുന്നു
 Name[mr]=Qt डिजाइनर मध्ये Ctrl+W ला Ctrl+F4 असे निश्चित करा
 Name[nb]=Viderefør Ctrl+W til Ctrl+F4 i QT Designer
-Name[nds]=Binnen Qt-Designer Strg+F4 na Strg+W toornen
+Name[nds]=Binnen Qt-Designer Strg+W as Strg+F4 översetten
 Name[ne]=Ctrl+W to Ctrl+F4 लाई Qt डिजाइनरमा पुन: नक्साङ्कन गर्नुहोस्
 Name[nl]=Ctrl+W omzetten naar Ctrl+F4 in Qt Designer
 Name[nn]=Omordna «Ctrl + W» til «Ctrl + F4» i Qt Designer
--- workspace/kwin/effects/blur.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kwin/effects/blur.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -83,7 +83,7 @@
 Comment[gu]=પાશ્વભાગને અર્ધ-પારદર્શક વિન્ડોઝમાં ઝાંખું કરે છે
 Comment[he]=טישטוש הרקע של חלונות שקופים למחצה
 Comment[hi]=अर्ध-पारदर्शी विंडो के पीछे पृष्ठभूमि को धुंधला करता है
-Comment[hne]=अल्पपारदर्सी विंडो के पिछोत ल धुंधला करथे 
+Comment[hne]=अल्पपारदर्सी विंडो के पिछोत अंगना ल धुंधला करथे 
 Comment[hsb]=Stwori młowojty pozadk za połpřewidnymi woknami.
 Comment[is]=Móðar bakgrunn aftan við hálfgegnsæa glugga
 Comment[it]=Sfoca lo sfondo dietro le finestre semitrasparenti
--- workspace/kwin/popupinfo.cpp	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kwin/popupinfo.cpp	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -51,6 +51,7 @@
     {
     setObjectName( name );
 
+    setWindowFlags( Qt::X11BypassWindowManagerHint );
     m_infoString = "";
     m_shown = false;
     reset();
@@ -140,7 +141,7 @@
     KSharedConfigPtr c(KGlobal::config());
     const KConfigGroup cg = c->group("PopupInfo");
     m_show = cg.readEntry("ShowPopup", false );
-    m_delayTime = cg.readEntry("PopupHideDelay", 350 );
+    m_delayTime = cg.readEntry("PopupHideDelay", 750 );
     }
 
 void PopupInfo::showInfo(const QString &infoString)
--- workspace/kwin/kwin.notifyrc	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kwin/kwin.notifyrc	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -3426,7 +3426,7 @@
 Name[gu]=વિન્ડો દૂર કરો
 Name[he]=מחק חלון
 Name[hi]=विंडो मिटाएँ
-Name[hne]=विंडो मिटाव
+Name[hne]=विंडो मेटाव
 Name[hr]=Izbriši prozor
 Name[hu]=Ablak törlése
 Name[is]=Eyða glugga
@@ -3501,7 +3501,7 @@
 Comment[gu]=વિન્ડો દૂર કરો
 Comment[he]=מחק חלון
 Comment[hi]=विंडो मिटाएँ
-Comment[hne]=विंडो मिटाव
+Comment[hne]=विंडो मेटाव
 Comment[hr]=Brisanje prozora
 Comment[hu]=Ablak törlése
 Comment[is]=Eyða glugga
@@ -5018,7 +5018,7 @@
 Name[gl]=Borrar o diálogo
 Name[gu]=સંવાદ દૂર કરો
 Name[hi]=संवाद मिटाएँ
-Name[hne]=बात ल मेटारव
+Name[hne]=बात ल मेटाव
 Name[hr]=Izbriši dijalog
 Name[hu]=Párbeszédablak eltűnése
 Name[is]=Eyða samskiptaglugga
--- workspace/kdm/kfrontend/sessions/xfce4.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/xfce4.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -40,7 +40,7 @@
 Comment[gu]=કોલ્સ્ટ્રેરોલ ફ્રી ડેસ્કટોપ એન્વાર્યમેન્ટ, આવૃત્તિ ૪. ડેસ્કટોપ એન્વાર્યમેન્ટ જે CDE ની યાદ અપાવે છે
 Comment[he]=‏Cholesterol Free Desktop Environment, גירסה 4. סביבת שולחן עבודה המזכירה את CDE
 Comment[hi]=कोलेस्ट्रॉल मुक्त डेस्कटॉप वातावरण, संस्करण 4. सीडीई की याद दिलाता एक डेस्कटॉप वातावरण
-Comment[hne]=कोलेस्ट्राल मुक्त डेस्कटाप वातावरन, संस्करन 4. सीडीई के याद दिलावत एक डेस्कटाप वातावरन
+Comment[hne]=कोलेस्ट्राल फ्री डेस्कटाप वातावरन, संस्करन 4. सीडीई के याद दिलावत एक डेस्कटाप वातावरन
 Comment[hr]=Cholesterol Free Desktop Environment verzija 4 - Okruženje radne površine koje podsjeća na CDE
 Comment[hsb]=Cholesterol Free Desktop Environment, wersija 4. System za dźěłowy powjerch, podobny na CDE
 Comment[hu]=The Cholesterol Free Desktop Environment, 4-es verzió. Egy CDE-szerű ablakkezelő
--- workspace/kdm/kfrontend/sessions/ctwm.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/ctwm.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -38,7 +38,7 @@
 Comment[gu]=ક્લાઉડેનું ટેબ વિન્ડો વ્યવસ્થાપક, TWM વર્ચ્યુઅલ સ્ક્રિન વગેરે વડે વિસ્તૃત કરેલ.
 Comment[he]=‏Claude's Tab Window Manager, ‏TWM עם שיפורים של מסכים וירטואליים וכו'
 Comment[hi]=क्लाउडे का टैब युक्त विंडो प्रबंधक, टीडबल्यूएम को आभासी स्क्रीन इत्यादि से बेहतर बनाया गया.
-Comment[hne]=क्लाउडे के टैब सहित विंडो प्रबंधक, टीडबल्यूएम ल आभासी स्क्रीन आदि से बेहतर बनाय गिस.
+Comment[hne]=क्लाउडे के टैब सहित विंडो प्रबंधक, टीडबल्यूएम ल आभासी स्क्रीन आदि से बेहतर बनाइस.
 Comment[hr]=Claudov Tab upravitelj prozora, TWM poboljšan virtualnim zaslonima, itd.
 Comment[hsb]=Claude's Tab Window Manager, TWM rozšěrjeny z wirtualnymi powjerchami atd.
 Comment[hu]=Claude lapozós ablakkezelője, lényegében a TWM, kiegészítve virtuális képernyőkkel, egyebekkel
--- workspace/kdm/kfrontend/sessions/oroborus.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/oroborus.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -42,7 +42,7 @@
 Comment[gu]=હલ્કું થીમ વાપરી શકાય તેવું વિન્ડો વ્યવસ્થાપક
 Comment[he]=מנהל חלונות קל משקל וניתן להתאמה אישית על ידי ערכות נושא
 Comment[hi]=हल्का, प्रसंगयोग्य विंडो प्रबंधक
-Comment[hne]=हल्का, प्रसंग बनाया जा सके अइसन विंडो प्रबंधक
+Comment[hne]=हल्का, प्रसंग बनाइस जा सके अइसन विंडो प्रबंधक
 Comment[hr]=Lagani upravitelj prozora s temama
 Comment[hsb]=Njenaročny woknowy rjadowar z temami 
 Comment[hu]=Kis erőforrásigényű ablakkezelő, témázási lehetőséggel
--- workspace/kdm/kfrontend/sessions/9wm.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/9wm.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -70,7 +70,7 @@
 Comment[pt]=Uma emulação do gestor de janelas do Plan 9 8-1/2
 Comment[pt_BR]=Uma emulação do gerenciador de janelas Plan 9 (8-1/2)
 Comment[ro]=Un emulator al managerului de ferestre 8-1/2 din Plan 9
-Comment[ru]=Эмуляция оконного менеджера Plan 9
+Comment[ru]=Эмуляция диспетчера окон Plan 9
 Comment[se]=Lásegieđahalli mii áddestaddá Plan 9 lásegieđahalli 8-1/2.
 Comment[sk]=Emulácia správcu okien 8-1/2 systému Plan 9
 Comment[sl]=Emulacija okenskega upravitelja Plan 9 8-1/2
--- workspace/kdm/kfrontend/sessions/ude.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/ude.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -38,7 +38,7 @@
 Comment[gu]=ધ UNIX ડેસ્કટોપ એન્વાર્યમેન્ટ
 Comment[he]=סביבת העבודה של יוניקס
 Comment[hi]=यूनिक्स डेस्कटॉप माहौल
-Comment[hne]=यूनिक्स डेस्कटाप माहौल
+Comment[hne]=यूनिक्स डेस्कटाप महिनाौल
 Comment[hr]=UNIX radno okruženje
 Comment[hsb]=UNIX Desktop Environment
 Comment[hu]=UNIX Desktop Environment ablakkezelő
--- workspace/kdm/kfrontend/sessions/xfce.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/xfce.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -38,7 +38,7 @@
 Comment[gu]=કોલ્સ્ટ્રેરોલ ફ્રી ડેસ્કટોપ એન્વાર્યમેન્ટ. ડેસ્કટોપ એન્વાર્યમેન્ટ જે CDE ની યાદ અપાવે છે
 Comment[he]=‏Cholesterol Free Desktop Environment. סביבת שולחן עבודה המזכירה את CDE
 Comment[hi]=कॉलेस्ट्रॉल रहित डेस्कटॉप माहौल. सीडीई जैसा एक डेस्कटॉप माहौल
-Comment[hne]=कालेस्ट्रोल रहित डेस्कटाप माहौल. सीडीई जइसन एक डेस्कटाप माहौल
+Comment[hne]=कालेस्ट्रोल रहित डेस्कटाप महिनाौल. सीडीई जइसन एक डेस्कटाप महिनाौल
 Comment[hr]=Cholesterol Free Desktop Environment - Okruženje radne površine koje podsjeća na CDE
 Comment[hsb]=Cholesterol Free Desktop Environment. System za dźěłowy powjerch, podobny na CDE
 Comment[hu]=Cholesterol Free Desktop Environment, egy a CDE-re emlékeztető ablakkezelő
--- workspace/kdm/kfrontend/sessions/waimea.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/waimea.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -42,7 +42,7 @@
 Comment[gu]=બ્લેકબોક્સ પર આધારિત ઉચ્ચ ગોઠવણી કરી શકાય તેવું વિન્ડો વ્યવસ્થાપક
 Comment[he]=מנהל חלונות מבוסס על Blackbox המאפשר התאמה אישית רבה
 Comment[hi]=ब्लेक-बाक्स आधारित, अत्यंत परिवर्तनीय विंडो प्रबंधक
-Comment[hne]=ब्लेक-बाक्स आधारित, अत्यंत बदलावीय विंडो प्रबंधक
+Comment[hne]=ब्लेक-बाक्स आधारित, बहुत परिवर्तनीय विंडो प्रबंधक
 Comment[hr]=Vrlo prilagodljiv upravitelj prozora zasnovan na Blackboxu
 Comment[hsb]=Jara konfigurujomny woknowy rjadowar,  bazowacy na Blackbox
 Comment[hu]=Egy sokféle beállítási lehetőséggel rendelkező ablakkezelő a Blackbox alapján
--- workspace/kdm/kfrontend/sessions/w9wm.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/w9wm.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -38,7 +38,7 @@
 Comment[gu]=9WM પર આધારિત વિન્ડો વ્યવસ્થાપક, જે વર્ચ્યુઅલ સ્ક્રિનો અને કીબોર્ડ જોડાણો વડે વિસ્તૃત કરી શકાય છે
 Comment[he]=מנהל חלונות המבוסס על 9WM, עם שיפורים לשולחנות עבודה וירטואליים ומיפויי מקשים
 Comment[hi]=9डबल्यूएम आधारित विंडो प्रबंधक, आभासी स्क्रीन तथा की-बोर्ड बाइंडिंग से बेहतर बनाया गया
-Comment[hne]=9डबल्यूएम आधार से विंडो प्रबंधक, आभासी स्क्रीन अउ की-बोर्ड बाइंडिंग से बेहतर बनाय गिस
+Comment[hne]=9डबल्यूएम आधार से विंडो प्रबंधक, आभासी स्क्रीन अउ की-बोर्ड बाइंडिंग से बेहतर बनाइस
 Comment[hr]=Upravitelj prozora zasnovan na 9WM, unaprijeđen virtualnim zaslonima i prečacima tipkovnice
 Comment[hsb]=Woknowy rjadowar bazowacy na 9WM, kiž podpěruje wirtuelne wobrazowki a tastaturowe definicije
 Comment[hu]=Egy 9WM-alapú ablakkezelő, virtuális képernyőkezeléssel, konfigurálható billentyűparancsokkal
--- workspace/kdm/kfrontend/sessions/cde.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/cde.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -37,7 +37,7 @@
 Comment[gu]=કોમન ડેસ્કટોપ એન્વાર્યમેન્ટ, ખાનગી ઔધોગિક પ્રમાણભૂત ડેસ્કટોપ એન્વાર્યમેન્ટ
 Comment[he]=‏Common Desktop Environment, סביבת שולחן עבודה קניינית סטנדרטית בתעשייה
 Comment[hi]=कॉमन डेस्कटॉप माहौल, एक स्वामित्व युक्त औद्योगिक मानक डेस्कटॉप माहौल
-Comment[hne]=कामन डेस्कटाप माहौल, एक मालिकाना वाले औद्योगिक मानक डेस्कटाप माहौल
+Comment[hne]=कामन डेस्कटाप महिनाौल, एक मालिकाना वाले औद्योगिक मानक डेस्कटाप महिनाौल
 Comment[hr]=Opće okruženje radne površine, standardizirano industrijskim vlasništvima
 Comment[hsb]=Common Desktop Environment, proprieterny dźěłowy powjerch po industrielnych normach
 Comment[hu]=The Common Desktop Environment, egy kereskedelmi, kváziszabványnak számító grafikus környezet
--- workspace/kdm/kfrontend/sessions/vtwm.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/vtwm.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -37,7 +37,7 @@
 Comment[gu]=વર્ચ્યુઅલ ટેબ વિન્ડો વ્યવસ્થાપક. વિઝ્યુલ સ્ક્રિન વગેરે વડે વિસ્તૃત બનાવેલ TWM.
 Comment[he]=‏Virtual Tab Window Manager. ‏TWM עם שיפורים לשולחנות עבודה וירטואליים וכו'
 Comment[hi]=आभासी टैब विंडो प्रबंधक. टीडबल्यूएम को आभासी स्क्रीन इत्यादि से बेहतर बनाया गया.
-Comment[hne]=आभासी टैब विंडो प्रबंधक. टीडबल्यूएम ल आभासी स्क्रीन आदि से बेहतर बनाय गिस.
+Comment[hne]=आभासी टैब विंडो प्रबंधक. टीडबल्यूएम ल आभासी स्क्रीन आदि से बेहतर बनाइस.
 Comment[hr]=Comment=Virtual Tab Window Manager. TWM poboljšan virtualnim zaslonima itd.
 Comment[hsb]=Virtual Tab Window Manager. TWM z wirtuelnymi wobrazowkami atd.
 Comment[hu]=Virtual Tab Window Manager, egy TWM-változat, támogatja a virtuális képernyőkezelést
--- workspace/kdm/kfrontend/sessions/sawfish.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/kdm/kfrontend/sessions/sawfish.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -41,7 +41,7 @@
 Comment[gu]=Emacs Lisp-જેવી ભાષાઓ વડે સ્ક્રિપ્ટ કરી શકાતું વિન્ડો વ્યવસ્થાપક
 Comment[he]=מנהל חלונות הניתן להרחבה ותכנות עם שפה דמויית Emacs Lisp
 Comment[hi]=ई-मेक्स लिस्प जैसे भाषा में स्क्रिप्ट किया जा सकने लायक विंडो प्रबंधक जिसे विस्तार दिया जा सकता है
-Comment[hne]=ई-मेक्स लिस्प जइसन भाखा मं स्क्रिप्ट कर सकीस जइसन विंडो प्रबंधक जऊन ल विस्तार दे जा सकथे 
+Comment[hne]=ई-मेक्स लिस्प जइसन भाखा मं स्क्रिप्ट कर सकिस जइसन विंडो प्रबंधक जऊन ल विस्तार दे जा सकथे 
 Comment[hr]=Proširivi upravitelj prozora pisan u skripti nalik na jezik Emacs Lisp
 Comment[hsb]=Rozšěrjomny woknowy rjadowar, kiž rozumi skriptowu rěč, podobnu na Emacs-Lisp
 Comment[hu]=Egy könnyen tovább bővíthető ablakkezelő, egy Emacs Lisp-szerű nyelvvel szkriptelhető
--- workspace/plasma/dataengines/CMakeLists.txt	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/dataengines/CMakeLists.txt	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -4,11 +4,9 @@
 add_subdirectory(favicons)
 add_subdirectory(filebrowser)
 add_subdirectory(hotplug)
-add_subdirectory(network)
 add_subdirectory(notifications)
 add_subdirectory(nowplaying)
 add_subdirectory(places)
-add_subdirectory(powermanagement)
 add_subdirectory(soliddevice)
 add_subdirectory(time)
 add_subdirectory(weather)
@@ -20,6 +18,8 @@
 
 if(NOT WIN32)
    add_subdirectory(mouse)
+   add_subdirectory(network)
+   add_subdirectory(powermanagement)
    add_subdirectory(systemmonitor)
    add_subdirectory(tasks)
 endif(NOT WIN32)
--- workspace/plasma/shells/desktop/panelview.cpp	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/shells/desktop/panelview.cpp	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -203,6 +203,11 @@
       m_triggerEntered(false)
 {
     Q_ASSERT(qobject_cast<Plasma::Corona*>(panel->scene()));
+
+    m_strutsTimer = new QTimer(this);
+    m_strutsTimer->setSingleShot(true);
+    connect(m_strutsTimer, SIGNAL(timeout()), this, SLOT(updateStruts()));
+
     if (panel) {
         connect(panel, SIGNAL(destroyed(QObject*)), this, SLOT(panelDeleted()));
         connect(panel, SIGNAL(toolBoxToggled()), this, SLOT(togglePanelController()));
@@ -248,10 +253,6 @@
     KWindowSystem::setOnAllDesktops(winId(), true);
 
     QTimer::singleShot(0, this, SLOT(init()));
-
-    m_strutsTimer = new QTimer(this);
-    m_strutsTimer->setSingleShot(true);
-    connect(m_strutsTimer,SIGNAL(timeout()),this,SLOT(updateStruts()));
 }
 
 PanelView::~PanelView()
--- workspace/plasma/containments/desktop/desktop.cpp	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/containments/desktop/desktop.cpp	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -77,7 +77,7 @@
     m_layout->setAutoWorkingArea(false);
     m_layout->setAlignment(Qt::AlignTop|Qt::AlignLeft);
     m_layout->setPlacementSpacing(20);
-    m_layout->setScreenSpacing(5);
+    m_layout->setScreenSpacing(0);
     m_layout->setShiftingSpacing(0);
     m_layout->setTemporaryPlacement(true);
     m_layout->setVisibilityTolerance(0.5);
--- workspace/plasma/applets/trash/plasma-applet-trash.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/applets/trash/plasma-applet-trash.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -86,7 +86,7 @@
 Comment[gu]=દૂર કરેલ વસ્તુઓનો અધિકાર
 Comment[he]=גישה לפריטים שנמחקו
 Comment[hi]=मिटाई जा चुकी वस्तुओं पर जाएँ
-Comment[hne]=मिटाय चीज मं पहुंच
+Comment[hne]=मेटाय चीज मं पहुंच
 Comment[hsb]=Přistup k wumazanym objektam
 Comment[is]=Aðgangur að hlutum sem hefur verið eytt
 Comment[it]=Accedi agli oggetti cancellati
--- workspace/plasma/applets/devicenotifier/test-predicate-openinwindow.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/applets/devicenotifier/test-predicate-openinwindow.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -27,7 +27,7 @@
 Name[gu]=ડોલ્ફિન સાથે ખોલો
 Name[he]=פתח באמצעות Dolphin
 Name[hi]=डॉल्फिन के साथ खोलें
-Name[hne]=डाल्फिन के साथ खोलव
+Name[hne]=डाल्फिन एखर साथ खोलव
 Name[hr]=Otvori pomoću Dolphina
 Name[hsb]=Z Dolphinom wočinić
 Name[hu]=Megnyitás a Dolphinnal
--- workspace/plasma/applets/quicklaunch/plasma-applet-quicklaunch.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/applets/quicklaunch/plasma-applet-quicklaunch.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -33,7 +33,7 @@
 Name[pt]=Lançamento Rápido
 Name[pt_BR]=Lançamento rápido
 Name[ro]=Lansare rapidă
-Name[ru]=Быстрый запуск
+Name[ru]=Панель приложений
 Name[sk]=Rýchle Spustenie
 Name[sl]=Hitri zagon
 Name[sr]=брзокрет
@@ -89,7 +89,7 @@
 Comment[pt]=Lança as suas aplicações favoritas
 Comment[pt_BR]=Executa os seus aplicativos favoritos
 Comment[ro]=Lansator pentru aplicațiile favorite
-Comment[ru]=Быстрый запуск приложений
+Comment[ru]=Панель с избранными приложениями
 Comment[si]=ඔබ කැමතීම යෙදුම් ආරම්ඹකය
 Comment[sk]=Rýchle spustenie obľúbených aplikácií
 Comment[sl]=Hitri zagon priljubljenih programov
--- workspace/plasma/applets/calendar/calendar.h	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/applets/calendar/calendar.h	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -44,6 +44,11 @@
          * The widget that displays the calendar.
          */
         QGraphicsWidget *graphicsWidget();
+
+    protected:
+        void updateDate();
+        void timerEvent(QTimerEvent *event);
+
     protected slots:
         void configAccepted();
 
@@ -52,6 +57,8 @@
 
         Plasma::Calendar *m_calendarDialog;
         Plasma::Svg *m_theme;
+        int m_date;
+        int m_updateTimerId;
 };
 
 K_EXPORT_PLASMA_APPLET(calendar, CalendarTest)
--- workspace/plasma/applets/calendar/calendar.cpp	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/applets/calendar/calendar.cpp	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -22,6 +22,7 @@
 
 #include <KDebug>
 #include <KIcon>
+#include <KSystemTimeZones>
 
 #include <Plasma/Svg>
 #include <Plasma/Theme>
@@ -32,14 +33,18 @@
 CalendarTest::CalendarTest(QObject *parent, const QVariantList &args)
     : Plasma::PopupApplet(parent, args),
     m_calendarDialog(0),
-    m_theme(0)
+    m_theme(0),
+    m_date(0),
+    m_updateTimerId(0)
 {
     setAspectRatioMode(Plasma::IgnoreAspectRatio);
+    setCacheMode(DeviceCoordinateCache);
 }
 
 void CalendarTest::init()
 {
     setPopupIcon("view-pim-calendar");
+    updateDate();
 }
 
 QGraphicsWidget *CalendarTest::graphicsWidget()
@@ -72,7 +77,6 @@
 
 void CalendarTest::paintIcon()
 {
-    //TODO: connect to a dataengine to repaint this thing on date change
     const int iconSize = qMin(size().width(), size().height());
 
     if (iconSize <= 0) {
@@ -96,7 +100,7 @@
     p.setPen(Plasma::Theme::defaultTheme()->color(Plasma::Theme::ButtonTextColor));
     font.setPixelSize(icon.size().height() / 2);
     p.setFont(font);
-    p.drawText(icon.rect().adjusted(0, icon.size().height()/4, 0, 0), Qt::AlignCenter, QString::number(QDate::currentDate().day()));
+    p.drawText(icon.rect().adjusted(0, icon.size().height()/4, 0, 0), Qt::AlignCenter, QString::number(m_date));
     m_theme->resize();
     p.end();
     setPopupIcon(icon);
@@ -107,4 +111,23 @@
     update();
 }
 
+void CalendarTest::updateDate()
+{
+    QDateTime d = QDateTime::currentDateTime();
+    m_date = d.date().day();
+    int updateIn = (24 * 60 * 60) - (d.toTime_t() + KSystemTimeZones::local().currentOffset()) % (24 * 60 * 60);
+    m_updateTimerId = startTimer(updateIn * 1000);
+    constraintsEvent(Plasma::FormFactorConstraint);
+}
+
+void CalendarTest::timerEvent(QTimerEvent *event)
+{
+    if (event->timerId() != m_updateTimerId) {
+        return;
+    }
+
+    killTimer(m_updateTimerId);
+    updateDate();
+}
+
 #include "calendar.moc"
--- workspace/plasma/applets/kickoff/ui/launcher.cpp	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/applets/kickoff/ui/launcher.cpp	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -313,7 +313,14 @@
         delete layout;
         layout = new QHBoxLayout();
         layout->addWidget(contentSwitcher);
-        layout->addWidget(contentArea);
+
+        QBoxLayout * contentLayout = new QVBoxLayout();
+        contentLayout->addWidget(contentAreaHeader);
+        contentLayout->addWidget(contentArea);
+        contentLayout->addWidget(contentAreaFooter);
+
+        layout->addItem(contentLayout);
+
         QBoxLayout * layout2 = new QVBoxLayout();
         if (tabOrder == NormalTabOrder) {
             layout2->addLayout(layout);
@@ -337,7 +344,14 @@
         QLayout * layout = q->layout();
         delete layout;
         layout = new QHBoxLayout();
-        layout->addWidget(contentArea);
+
+        QBoxLayout * contentLayout = new QVBoxLayout();
+        contentLayout->addWidget(contentAreaHeader);
+        contentLayout->addWidget(contentArea);
+        contentLayout->addWidget(contentAreaFooter);
+
+        layout->addItem(contentLayout);
+
         layout->addWidget(contentSwitcher);
         QBoxLayout * layout2 = new QVBoxLayout();
         if (tabOrder == NormalTabOrder) {
--- workspace/plasma/applets/CMakeLists.txt	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/applets/CMakeLists.txt	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -1,11 +1,9 @@
 add_subdirectory(analog-clock)
 add_subdirectory(activitybar)
-add_subdirectory(battery)
 add_subdirectory(calendar)
 add_subdirectory(devicenotifier)
 add_subdirectory(digital-clock)
 add_subdirectory(icon)
-add_subdirectory(kickoff)
 add_subdirectory(lock_logout)
 add_subdirectory(quicklaunch)
 add_subdirectory(trash)
@@ -15,6 +13,8 @@
 endif(QT_QTWEBKIT_FOUND)
 
 if(NOT WIN32)
+   add_subdirectory(battery)
+   add_subdirectory(kickoff)
    add_subdirectory(pager)
    add_subdirectory(system-monitor)
    add_subdirectory(systemtray)
--- workspace/plasma/scriptengines/ruby/ChangeLog	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/scriptengines/ruby/ChangeLog	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -1,3 +1,8 @@
+2009-02-10  Richard Dale  <richard.j.dale@gmail.com>
+* For the methods that need to be special cased to work with 
+  PlasmaScripting::Applet type arguments, add lower case/underscore versions
+  of the methods
+
 2009-02-05  Richard Dale  <richard.j.dale@gmail.com>
 * Added various special casing to the Plasma::ToolTipManager class so you can
   pass a PlasmaScripting::Applet to the methods. Thanks to Mathieu Jobin for
--- workspace/plasma/scriptengines/ruby/applet.rb	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/scriptengines/ruby/applet.rb	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -498,6 +498,10 @@
       end
     end
 
+    def set_content(widget, data)
+      setContent(widget, data)
+    end
+
     def clearContent(widget)
       if widget.kind_of?(PlasmaScripting::Applet)
         super(widget.applet_script.applet)
@@ -506,6 +510,10 @@
       end
     end
 
+    def clear_content(widget)
+      clearContent(widget)
+    end
+
     def show(widget)
       if widget.kind_of?(PlasmaScripting::Applet)
         super(widget.applet_script.applet)
@@ -534,6 +542,10 @@
       end
     end
 
+    def register_widget(widget)
+      registerWidget(widget)
+    end
+
     def unregisterWidget(widget)
       if widget.kind_of?(PlasmaScripting::Applet)
         super(widget.applet_script.applet)
@@ -541,6 +553,10 @@
         super
       end
     end
+
+    def unregister_widget(widget)
+      unregisterWidget(widget)
+    end
   end
 
   class TreeView < Qt::Base
@@ -575,6 +591,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -601,6 +621,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -627,6 +651,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -653,6 +681,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -679,6 +711,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -705,6 +741,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -731,6 +771,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -757,6 +801,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -783,6 +831,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -809,6 +861,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -835,6 +891,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -861,6 +921,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -887,6 +951,10 @@
       end
     end
 
+    def parent_item=(item)
+      setParentItem(item)
+    end
+
     def parentItem=(item)
       setParentItem(item)
     end
@@ -913,6 +981,10 @@
       end
     end
 
+    def add_item(*args)
+      addItem(*args)
+    end
+
     def addItem(*args)
       sargs = []
       for i in 0...args.length do
@@ -933,6 +1005,10 @@
       end
     end
 
+    def set_alignment(item, alignment)
+      setAlignment(item, alignment)
+    end
+
     def setAlignment(item, alignment)
       if item.kind_of?(PlasmaScripting::Applet)
         super(item.applet_script.applet, alignment)
@@ -955,6 +1031,10 @@
       super(*sargs)
     end
 
+    def add_item(*args)
+      addItem(*args)
+    end
+
     def addItem(*args)
       sargs = []
       for i in 0...args.length do
@@ -975,6 +1055,10 @@
       end
     end
 
+    def insert_item(index, item)
+      insertItem(index, item)
+    end
+
     def insertItem(index, item)
       if item.kind_of?(PlasmaScripting::Applet)
         super(index, item.applet_script.applet)
@@ -983,6 +1067,10 @@
       end
     end
 
+    def set_alignment(item, alignment)
+      setAlignment(item, alignment)
+    end
+
     def setAlignment(item, alignment)
       if item.kind_of?(PlasmaScripting::Applet)
         super(item.applet_script.applet, alignment)
@@ -991,6 +1079,10 @@
       end
     end
 
+    def set_stretch_factor(item, stretch)
+      setStretchFactor(item, stretch)
+    end
+
     def setStretchFactor(item, stretch)
       if item.kind_of?(PlasmaScripting::Applet)
         super(item.applet_script.applet, stretch)
@@ -999,6 +1091,10 @@
       end
     end
 
+    def stretch_factor(item)
+      stretchFactor(item)
+    end
+
     def stretchFactor(item)
       if item.kind_of?(PlasmaScripting::Applet)
         super(item.applet_script.applet)
--- workspace/plasma/scriptengines/google_gadgets/plasma-scriptengine-googlegadgets.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/scriptengines/google_gadgets/plasma-scriptengine-googlegadgets.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -70,6 +70,7 @@
 Comment[pl]=Gadżety Google na pulpicie
 Comment[pt]='Gadgets' do Google Desktop
 Comment[pt_BR]=Gadgets do Google Desktop
+Comment[ro]=Gadgeturi de birou Google
 Comment[ru]=Гаджеты Google для рабочего стола
 Comment[sl]=Gradniki Google Gadgets
 Comment[sr]=Гуглове справице за површ
--- workspace/plasma/scriptengines/google_gadgets/plasma-packagestructure-googlegadgets.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/scriptengines/google_gadgets/plasma-packagestructure-googlegadgets.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -84,6 +84,7 @@
 Comment[pl]=Gadżetu Google na pulpicie
 Comment[pt]='Gadget' do Google Desktop
 Comment[pt_BR]=Gadget do Google Desktop
+Comment[ro]=Gadget de birou Google
 Comment[ru]=Гаджет Google для рабочего стола
 Comment[sl]=Gradnik Google Gadget
 Comment[sr]=Гуглове справице за површ
--- workspace/plasma/scriptengines/qedjescript/plasma-appletscript-qedje.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/scriptengines/qedjescript/plasma-appletscript-qedje.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -39,6 +39,7 @@
 Comment[pl]=Gadżety QEdje
 Comment[pt]=Itens do QEdje
 Comment[pt_BR]=Gadgets do QEdje
+Comment[ro]=Gadgeturi QEdje
 Comment[ru]=Гаджеты QEdje
 Comment[sl]=Gradniki QEdje
 Comment[sr]=Куедје справице
--- workspace/plasma/scriptengines/qedjescript/plasma-packagestructure-qedje.desktop	(.../tags/KDE/4.2.1/kdebase)	(wersja 935993)
+++ workspace/plasma/scriptengines/qedjescript/plasma-packagestructure-qedje.desktop	(.../branches/KDE/4.2/kdebase)	(wersja 935993)
@@ -31,6 +31,7 @@
 Name[pl]=Gadżety QEdje
 Name[pt]=Itens do QEdje
 Name[pt_BR]=Gadgets do QEdje
+Name[ro]=Gadgeturi QEdje
 Name[ru]=Гаджеты QEdje
 Name[sl]=Gradniki QEdje
 Name[sr]=Куедје справице
@@ -78,6 +79,7 @@
 Comment[pl]=Gadżet QEdje
 Comment[pt]=Item do QEdje
 Comment[pt_BR]=Gadget do QEdje
+Comment[ro]=Gadget QEdje
 Comment[ru]=Гаджет QEdje
 Comment[sl]=Gradnik QEdje
 Comment[sr]=КуЕдје справица
