--- CMakeLists.txt.orig	2010-01-08 05:34:31.410760301 +0100
+++ CMakeLists.txt	2010-01-08 05:35:34.651532005 +0100
@@ -60,10 +60,7 @@
 endif(PYQT4_FOUND AND PYQT4_VERSION STRLESS "040400")
 
 macro_optional_find_package(PyKDE4)
-set(PythonSupport_FOUND FALSE)
-if(PYTHONLIBRARY_FOUND AND SIP_FOUND AND PYQT4_FOUND AND PYKDE4_FOUND)
-  set(PythonSupport_FOUND TRUE)
-endif(PYTHONLIBRARY_FOUND AND SIP_FOUND AND PYQT4_FOUND AND PYKDE4_FOUND)
+set(PythonSupport_FOUND TRUE)
 
 macro_log_feature(PythonSupport_FOUND "Python" "KDE Python support" "http://techbase.kde.org/Development/Languages/Python" FALSE "" "Needed to build plasma scriptengine for Python")
 #### End Python support ####
