--- kde-workspace-4.8.2/powerdevil/daemon/backends/upower/powerdevilupowerbackend.h~	2012-03-30 00:02:31.000000000 +0200
+++ kde-workspace-4.8.2/powerdevil/daemon/backends/upower/powerdevilupowerbackend.h	2012-04-19 23:54:03.961628717 +0200
@@ -70,6 +70,7 @@
 
     // brightness
     float m_cachedBrightness;
+    QElapsedTimer m_cachedBrightness_timer;
     XRandrBrightness         *m_brightnessControl;
     OrgFreedesktopUPowerInterface *m_upowerInterface;
     OrgFreedesktopUPowerKbdBacklightInterface *m_kbdBacklight;
--- kde-workspace-4.8.2/powerdevil/daemon/backends/upower/powerdevilupowerbackend.cpp~	2012-03-30 00:02:31.000000000 +0200
+++ kde-workspace-4.8.2/powerdevil/daemon/backends/upower/powerdevilupowerbackend.cpp	2012-04-19 23:56:23.768337437 +0200
@@ -108,6 +108,7 @@
 
     if (!controls.isEmpty()) {
         m_cachedBrightness = brightness(Screen);
+	m_cachedBrightness_timer.start();
         kDebug() << "current screen brightness: " << m_cachedBrightness;
     }
 
@@ -153,6 +154,9 @@
 
 void PowerDevilUPowerBackend::brightnessKeyPressed(PowerDevil::BackendInterface::BrightnessKeyType type)
 {
+    if (m_cachedBrightness_timer.elapsed() < 100)
+        return; // two key pressed delivered at once; 100ms marign
+
     BrightnessControlsList controls = brightnessControlsAvailable();
     QList<QString> screenControls = controls.keys(Screen);
 
@@ -174,11 +178,14 @@
             newBrightness = brightness(Screen);
             if (!qFuzzyCompare(newBrightness, m_cachedBrightness)) {
                 m_cachedBrightness = newBrightness;
+                m_cachedBrightness_timer.restart();
                 onBrightnessChanged(Screen, m_cachedBrightness);
             }
         }
     } else {
         m_cachedBrightness = currentBrightness;
+        m_cachedBrightness_timer.restart();
+        onBrightnessChanged(Screen, m_cachedBrightness);
     }
 }
 
